src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java

```diff
@@ -216,173 +429,169 @@ private static void deleteProjectData(@Nonnull final SphereClient client) {
     deleteProductTypes(client);
   }
 
-  private static void deleteProductTypes(@Nonnull final SphereClient ctpClient) {
+  private static void deleteProductTypes(@Nonnull final ProjectApiRoot ctpClient) {
     deleteProductTypeAttributes(ctpClient);
     deleteProductTypesWithRetry(ctpClient);
   }
 
-  private static void deleteProductTypesWithRetry(@Nonnull final SphereClient ctpClient) {
-    final Consumer<List<ProductType>> pageConsumer =
-        pageElements ->
-            CompletableFuture.allOf(
-                    pageElements.stream()
-                        .map(productType -> deleteProductTypeWithRetry(ctpClient, productType))
-                        .map(CompletionStage::toCompletableFuture)
-                        .toArray(CompletableFuture[]::new))
-                .join();
-
-    CtpQueryUtils.queryAll(ctpClient, ProductTypeQuery.of(), pageConsumer)
+  private static void deleteProductTypesWithRetry(@Nonnull final ProjectApiRoot ctpClient) {
+    // todo: add retry
```

Comment:
Is there a ticket for this?

Reply:
https://commercetools.atlassian.net/browse/DEVX-276